core/src/main/java/com/opensymphony/xwork2/config/impl/DefaultConfiguration.java

```diff
@@ -395,6 +396,7 @@ public static ContainerBuilder bootstrapFactories(ContainerBuilder builder) {
                 .factory(SecurityMemberAccess.class, Scope.PROTOTYPE)
                 .factory(OgnlGuard.class, StrutsOgnlGuard.class, Scope.SINGLETON)
                 .factory(ProviderAllowlist.class, Scope.SINGLETON)
+                .factory(ThreadAllowlist.class, Scope.SINGLETON)
```

Comment:
Wouldn't be better to use `PROTOTYPE` scope and do not use ThreadLocal instead?

Reply:
This bean was introduced as a way to mutate the allowlist read by `SecurityMemberAccess` - if it were `prototype` scope, `ParametersInterceptor` and `SecurityMemberAccess` would not share the same instance of `ThreadAllowlist` and thus this wouldn't be possible.

Reply:
👍